src/panels/config/lovelace/resources/dialog-lovelace-resource-detail.ts

```diff
@@ -23,6 +23,17 @@ import { haStyleDialog } from "../../../../resources/styles";
 import { HomeAssistant } from "../../../../types";
 import { LovelaceResourceDetailsDialogParams } from "./show-dialog-lovelace-resource-detail";
 
+function detectResourceType(url: string) {
+  const ext = url.split(".").pop() || "";
+  const styleSheetExt = ["css", "scss", "sass"];
```

Comment:
This should be outside of the function. No need to create it everytime the function is ran